stl/src/winapisupp.cpp

```diff
@@ -14,11 +14,36 @@
 #pragma warning(disable : 4265) // non-virtual destructor in base class
 #include <wrl/wrappers/corewrappers.h>
 #pragma warning(pop)
-#include <cstdint>
+
+#if !defined(_ONECORE)
+namespace {
+
+    enum wrapKERNEL32Functions {
+#if !defined(_CRT_WINDOWS) && !defined(UNDOCKED_WINDOWS_UCRT)
+        eGetCurrentPackageId,
+#endif // !defined(_CRT_WINDOWS) && !defined(UNDOCKED_WINDOWS_UCRT)
+
+#if _STL_WIN32_WINNT < _WIN32_WINNT_WIN8
+        eGetSystemTimePreciseAsFileTime,
+#endif // _STL_WIN32_WINNT < _WIN32_WINNT_WIN8
+
+        eMaxKernel32Function
+    };
+
+    PVOID __KERNEL32Functions[eMaxKernel32Function > 0 ? eMaxKernel32Function : 1]{};
+
+// Use this macro for caching a function pointer from a DLL
+#define STOREFUNCTIONPOINTER(instance, function_name) \
+    __KERNEL32Functions[e##function_name] = reinterpret_cast<PVOID>(GetProcAddress(instance, #function_name))
+
+// Use this macro for retrieving a cached function pointer from a DLL
+#define IFDYNAMICGETCACHEDFUNCTION(name) \
+    if (const auto pf##name = reinterpret_cast<decltype(&name)>(__KERNEL32Functions[e##name]); pf##name)
+

```

Comment:
now that we are only doing this a few times it might be easier to just expand these macros.

If we're going to use macros for this we could also use xmacros to define the above enumeration and the below initialization function at the same time.